homeassistant/components/cast/media_player.py

```diff
@@ -366,23 +425,43 @@ def async_cast_discovered(discover: ChromecastInfo):
     async def async_set_cast_info(self, cast_info):
         """Set the cast information and set up the chromecast object."""
         import pychromecast
-        old_cast_info = self._cast_info
         self._cast_info = cast_info
 
+        if cast_info.service not in self.services:
+            _LOGGER.debug("[%s %s (%s:%s)] Got new service: %s (%s)",
+                          self.entity_id, self._cast_info.friendly_name,
+                          self._cast_info.host, self._cast_info.port,
+                          cast_info.service, self.services)
+
+        self.services.add(cast_info.service)
+
         if self._chromecast is not None:
-            if old_cast_info.host_port == cast_info.host_port:
-                _LOGGER.debug("No connection related update: %s",
-                              cast_info.host_port)
-                return
-            await self._async_disconnect()
+            # Only setup the chromecast once, added elements to services
+            # will automatically be picked up.
+            return
 
         # pylint: disable=protected-access
-        _LOGGER.debug("Connecting to cast device %s", cast_info)
-        chromecast = await self.hass.async_add_job(
-            pychromecast._get_chromecast_from_host, (
-                cast_info.host, cast_info.port, cast_info.uuid,
-                cast_info.model_name, cast_info.friendly_name
-            ))
+        if len(self.services) == 1 and None in self.services:
```

Comment:
If the chromecast has been setup from configuration.yaml, the service is `None` and we end up here.

Reply:
Can you add this as a comment in the code?

Reply:
Also, wouldn't it make more sense to protect against `None` being added to services ?

Reply:
When writing it, the idea was that `None` would mean manually configured, maybe it didn't turn out too well.

Reply:
I rewrote it a bit such that self.services is None for manually added Casts, not sure if it's better.

Reply:
Btw, updating service info for a manually added cast is blocked. I think that's OK though, since it has been explicitly added by hostname / IP?

Reply:
Yeah